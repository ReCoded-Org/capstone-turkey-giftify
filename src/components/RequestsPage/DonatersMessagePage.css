.donatersMessageContainer {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-evenly;
  align-items: stretch;
  align-content: stretch;
  width: 100%;
}

.mainProductDetails {
  background: #90e0ef;
  border-radius: 15px;
  width: 45%;
}

.productDetails {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-around;
  align-items: stretch;
  align-content: stretch;
  margin: 5% 0%;
}

.productDetailsParentDiv {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}
.productDetailsChildDiv {
  width: 85%;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: center;
  align-content: stretch;
}

.productDetailsChildDiv > span {
  font-weight: bold;
  width: 100%;
}

.productDetailsParentDiv > p {
  background: #f8f9fa;
  border: 1px solid #006eff;
  box-sizing: border-box;
  border-radius: 15px;
  padding: 3%;
  width: 90%;
}

.productDetailsImage {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-around;
  align-items: center;
  align-content: stretch;
  width: 100%;
}

.productDetailsImage > img {
  width: 75%;
  border-radius: 15px;
  margin: 0% 0% 5% 0%;
}

.messageArea {
  background: #90e0ef;
  border-radius: 15px;
  width: 45%;
}

.donatersMessageText {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-around;
  align-items: baseline;
  align-content: stretch;
  width: 100%;
  margin: 5%;
}

.donatersMessageText > span {
  margin: 0% 0% 1% 2%;
  font-weight: bold;
}

.donatersMessageText > p {
  background: #f8f9fa;
  border: 1px solid #006eff;
  box-sizing: border-box;
  border-radius: 15px;
  padding: 3%;
  width: 90%;
}

.productDetailsChildDiv > svg,
.productDetailsImage > svg,
.productDeleteParentDiv > svg {
  cursor: pointer;
}

.productDeleteParentDiv > svg {
  margin-top: 4%;
  font-size: 30px;
}

.productDeleteParentDiv > .productDetailsChildDiv,
.productDeleteParentDiv {
  text-align: center;
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  align-content: stretch;
}

.productDeleteParentDiv > .productDetailsChildDiv > span {
  font-size: 25px;
}

.productDeleteParentDiv:hover {
  border-radius: 50px;
  background-image: linear-gradient(
    to top,
    #dbdcd7 0%,
    #dddcd7 24%,
    #e2c9cc 30%,
    #e7627d 46%,
    #b8235a 59%,
    #801357 71%,
    #3d1635 84%,
    #1c1a27 100%
  );
  transition: 0.3s;
  color: #ffffff;
}

@media only screen and (min-width: 120px) and (max-width: 531px) {
  .donatersMessageContainer {
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    justify-content: space-around;
    align-items: center;
    align-content: stretch;
  }

  .mainProductDetails {
    width: 85%;
  }

  .messageArea {
    margin-top: 5%;
    width: 85%;
  }
}
