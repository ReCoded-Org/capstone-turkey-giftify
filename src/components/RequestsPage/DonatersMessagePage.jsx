import React, { useState, useEffect } from "react";
import { Card } from "react-bootstrap";
import "./DonatersMessagePage.css";
import { useParams } from "react-router";
import db from "./../../firebase";

const DonatersMessagePage = () => {
  const { id } = useParams();

  const [usersMessages, setUsersMessages] = useState([]);
  const [productDetails, setProductDetails] = useState([]);
  let categoriesName = [];

  const getRequestsMessage = async () => {
    const categoriesPath = db.collection("categories");
    await categoriesPath.get().then((querySnapshot) => {
      querySnapshot.forEach((doc) => {
        categoriesName.push(doc.id);
        categoriesName.map((category) => {
          return categoriesPath
            .doc(`${category}`)
            .collection("items")
            .doc(`${id}`)
            .collection("requests")
            .get()
            .then((querySnapshot) => {
              querySnapshot.forEach((doc) => {
                console.log("docc: ", doc.data());
                setUsersMessages((prevState) => {
                  return {
                    ...prevState,
                    [doc.id]: [doc.data().reason, doc.data().userId],
                  };
                });
              });
            });
        });
      });
    });
  };

  const getProductDetails = async () => {
    const categoriesPath = db.collection("categories");
      categoriesName.map((category) => {
        return categoriesPath
        // use category
          .doc(`${category}`)
          .collection("items")
          .doc(`${id}`)
          .get()
          .then((querySnapshot) => {
            console.log("querySnapshot.data(): ", querySnapshot.data());
            setProductDetails(querySnapshot.data());
          });
      });
  };

  console.log("usersMessages: ", usersMessages);
  console.log("productDetails: ", productDetails);

  useEffect(() => {
    getRequestsMessage();
    getProductDetails();
    // eslint-disable-next-line
  }, []);

  return (
    <div className="donatersMessageContainer">
      <div className="mainProductDetails">
        <p>SADECE CIGLIK ATIYORUM ARTIK</p>
        {/* returns undefined state and after an action call returns the state correctly */}
        {/* {Object.values(productDetails).map((info) => (
                    <Card key={info}>

                    </Card>
                ))} */}
      </div>
      <div className="messageArea">
        {Object.values(usersMessages).map((info) => (
          <div className="donatersMessageText">
            <span>User ID: {info[1]}</span>
            <p>{info[0]}</p>
          </div>
        ))}
      </div>
    </div>
  );
};

export default DonatersMessagePage;
