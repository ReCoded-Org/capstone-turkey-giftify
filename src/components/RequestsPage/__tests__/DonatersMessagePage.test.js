import React from "react";
import DonatersMessagePage from "./../DonatersMessagePage";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: jest
    .fn()
    .mockReturnValue({ category: "clothing", id: "3iz39LBgqzm67DAOhPy1" }),
}));

it("DonatersMessagePage rendered correctly", () => {
  const tree = renderer.create(<DonatersMessagePage />).toJSON();
  expect(tree).toMatchSnapshot();
});
