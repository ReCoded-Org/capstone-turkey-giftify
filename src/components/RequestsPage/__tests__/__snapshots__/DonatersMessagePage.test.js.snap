// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonatersMessagePage rendered correctly 1`] = `
<div
  className="donatersMessageContainer"
>
  <div
    className="mainProductDetails"
  />
  <div
    className="messageArea"
  />
</div>
`;
